api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1361,30 +1367,33 @@ public void getUserByEmail_shouldGetUserByUsingEmail() {
 		User user = userService.getUserByEmail("hank.williams@gmail.com");
 		assertNotNull("User with email hank.williams@gmail not found in database", user);
 	}
-	/*
-	 * @see UserService#getUserByActivationKey(String)
-	 */
+	
 	@Test
-	public void getUserByActivationKey_shouldGetUserUsingActivationKey() {
-		executeDataSet(XML_FILENAME);
-		String key="XbklsieiskeuztecEnPf";
-		User user = userService.getUserByActivationKey(key);	
-		assertNotNull("User with activationKey ", user); 
-		String[] tokens = user.getActivationKey().split(":");
-		assertEquals(key,tokens[0]);		
-
+	public void setUserActivationKey_shouldCreateUserActivationKey() {
+		User u = new User();
+		u.setPerson(new Person());
+		u.addName(new PersonName("Benjamin", "A", "Wolfe"));
+		u.setUsername("bwolfe");
+		u.getPerson().setGender("M");
+		User createdUser = userService.createUser(u, "Openmr5xy");
+		assertEquals(createdUser, userService.setUserActivationKey(createdUser));
 	}
 	
-	/**
-	 * @see UserService#getUserByActivationKey(String)
-	 */
-	@Test
-	public void getUserByActivationKey_shouldReturnNullIfNoUserFoundWithGivenKey() {
-		executeDataSet(XML_FILENAME);
-		String key="XbklsieiskNoMatchKey";
-		User user = userService.getUserByActivationKey(key);			
-		assertNull(user);
+	@Test 
+	public void getUserByActivationKey_shouldGetUserByActivationKey(){
+		User u = new User();
+		u.setPerson(new Person());
+		u.addName(new PersonName("Benjamin", "A", "Wolfe"));
+		u.setUsername("bwolfe");
+		u.getPerson().setGender("M");
+		User createdUser = userService.createUser(u, "Openmr5xy");
+		String key="h4ph0fpNzQCIPSw8plJI";
+		int validTime = 10*60*1000; //equivalent to 10 minutes for token to be valid
+		Long tokenTime = System.currentTimeMillis() + validTime;
+		LoginCredential credentials = dao.getLoginCredential(createdUser);
```

Comment:
This is actually not the right thing to do, instead call setActivationKey and make sure getUserByActivation key returns the same user